core-bundle/src/Resources/contao/templates/forms/form_range.html5

```diff
@@ -17,5 +17,5 @@
     <p class="error"><?= $this->getErrorAsString() ?></p>
   <?php endif; ?>
 
-  <input type="<?= $this->type ?>" name="<?= $this->name ?>" id="ctrl_<?= $this->id ?>" class="range<?php if ($this->class): ?> <?= $this->class ?><?php endif; ?>"<?php if ($this->value): ?> value="<?= (float) $this->value ?>"<?php endif; ?><?= $this->getAttributes() ?>>
+  <input type="<?= $this->type ?>" name="<?= $this->name ?>" id="ctrl_<?= $this->id ?>" class="range<?php if ($this->class): ?> <?= $this->class ?><?php endif; ?>"<?php if ('' != $this->value): ?> value="<?= (float) $this->value ?>"<?php endif; ?><?= $this->getAttributes() ?>>
```

Comment:
I think we should use strict comparison (`!==`) in this case.
```suggestion
  <input type="<?= $this->type ?>" name="<?= $this->name ?>" id="ctrl_<?= $this->id ?>" class="range<?php if ($this->class): ?> <?= $this->class ?><?php endif; ?>"<?php if ('' !== $this->value): ?> value="<?= (float) $this->value ?>"<?php endif; ?><?= $this->getAttributes() ?>>
```

Reply:
We are casting the value to `(float)` but we use `'' !==` to compare it with an empty string. Is this intended?

Reply:
I would say this is correct - value can be an int or float and should only be set if the input field contains an entry, which can also be 0

Reply:
> Is this intended?

I think so, yes. The `value` attribute should always be a number, that’s why we cast to `(float)`. The `'' !==` is used to differentiate between int/float `0` (in which case we want to output a value) and the empty string (which we do not want to output).

Reply:
What if `$this-value` is `null` or `false`? The if condition would be true in this case and the attribute would be set to `value="0"`. IMHO, it should be `'' !== (string) $this->value`, shouldn't it?

Reply:
when the form widget is created $this->value is the default value - current its not possible to set 0 (zero)!